Kernel 6.12.x

ChrisIgel opened the pull request:
**- What I did**

- Add a the new kernel series 6.12.x
- Enable AppArmor
- Set EFI stub properties

**- How I did it**
Followed the instructions.

**- How to verify it**
Run the tests and verify BIOS build/boot still works for all three architectures.

**- Description for the changelog**
Add a the new kernel series 6.12.x


I also had to update the kernel `Dockerfile` and either copy the `Image.gz` (for the 6.6.x) or the `vmlinuz.efi` (for the 6.12.x). I hope/think the `vmlinuz.efi` is the right one. I didn't have a chance to try it out yet. Supposedly, it also works with non-UEFI systems.

deitch commented on kernel/Dockerfile:
```diff
@@ -140,10 +140,12 @@ RUN make -j "$(getconf _NPROCESSORS_ONLN)" KCFLAGS="-fno-pie" && \
     aarch64) \
-        cp arch/arm64/boot/Image.gz /out/kernel; \
+        cp arch/arm64/boot/Image.gz /out/kernel || true; \
+        cp arch/arm64/boot/vmlinuz.efi /out/kernel || true; \
```
In any case, we definitely want it to fail if it finds none of them!

deitch commented:
Oh and needs to be signed, DCO check is failing

ChrisIgel commented:
I fixed it.

Regarding https://github.com/linuxkit/linuxkit/pull/4167: I'm happy to also include it.

ChrisIgel commented:
I included the change. We didn't connect yet, but I will ping you sometime in the next days.

deitch commented:
So this is ready? I can try and build and push out?

deitch commented:
Ah, it is rebuilding `pkg/firmware`. Well, might as well let it run.

deitch commented:
And the commit checked out is not quite the same, look [here](https://github.com/linuxkit/linuxkit/actions/runs/18662018329/job/53496189474#step:2:87):

```
  /usr/bin/git checkout --progress --force refs/remotes/pull/4185/merge
  Note: switching to 'refs/remotes/pull/4185/merge'.
  
  You are in 'detached HEAD' state. You can look around, make experimental
  changes and commit them, and you can discard any commits you make in this
  state without impacting any branches by switching back to a branch.
  
  If you want to create a new branch to retain commits you create, you may
  do so (now or later) by using -c with the switch command. Example:
  
    git switch -c <new-branch-name>
  
  Or undo this operation with:
  
    git switch -
  
  Turn off this advice by setting config variable advice.detachedHead to false
  
  HEAD is now at 78fe595 Merge e67beb36dbb749cf012b464c0b77100dde2980a8 into d94f2e90954fed091950d734e61f6452a787f17a
/usr/bin/git log -1 --format=%H
78fe595e14a5efa152748b8bd30ae1b0bd726649
```

That is not the most recent commit. I actually do not see it anywhere.

deitch commented:
OK, that is the synthetic merge commit. I was able to replicate it locally, my HEAD now points at that commit. But I _still_ have no issue with the `make -C kernel update-kernel-yamls`, which changes nothing. So why does it do so in the Actions?

deitch commented:
In the meantime, despite my (clearly erroneous) comments earlier, can you put back the hash changes on 6.6.71 in the few places it was there? I would guess that `make -C kernel update-kernel-yamls` would do it.

ChrisIgel commented:
I fixed the script (I think) and also updated the hashes. Can you verify this is correct now? Also is it expected that the kernel test hashes do not get updated automatically by the `update-kernel-yamls` (also not the 6.12.x ones)?

ChrisIgel commented:
I removed the commit. It at least fixes these errors I get
```
rm: missing operand
Try 'rm --help' for more information.
rm: missing operand
Try 'rm --help' for more information.
```
and also from what I understand the gnu sed does not have the `-E` flag only the `-r`.

deitch commented:
I think it still is missing those 3 or so changes for 6.6.71-<hash>

deitch commented:
Just update the linked script in the [test package](https://github.com/linuxkit/linuxkit/tree/master/test/pkg/kernel-config). It is a bunch of simple tests, some of which are in `if-then` statements. Just update it so that those are different for 6.12.x (or ge 6.12.x)

ChrisIgel commented:
I pushed a commit that should fix everything except the cpuset thing. I'm not sure how to fix it yet but I'm on it.

deitch commented:
As for cpuset, [this issue](https://github.com/raspberrypi/linux/issues/6396) may provide insight (there are others like it).

I think I remove that universal check in `check-kernel-config.sh` for `cpuset`, and place it in your conditional line just after. (the same as for `ntfs`).

deitch commented:
CI looks happy. Huzzah!
